### DISCUSSION TITLE
Spatial covariate (factor) fixed effect

### DISCUSSION THREAD

**Discussion by orrortega** [inlabru-org/inlabru#185] [Q&A] [answered]
Hi,

I have been working with inla and inlabru for a while and now I am working in a new problem that I don't know how to solve.
I am working with preferencial sampling data with two sources (transects and traps) I have added spatial covariates such as slope and aspect without any problem. However, when I include a spatial covariate with levels - in my case vegetation, I am not able to added as a fixed effect.
I have tried first with all the points together, with a simple cp and the only output I am able to obtain is with the vegetation as a random effect but not as fixed effect.

Please, could you give some advice.

Thanks

Oscar

**Comment by finnlindgren** [comment, marked as answer]
Have you looked at the https://inlabru-org.github.io/inlabru/articles/web/2d_lgcp_covars.html example? For factor variables, one needs to use one of the factor_* component models (that internally appear as “random” in the inla output for technical reasons, but they use a fixed precision parameter exactly like regular “fixed” effects do).

**Reply by orrortega** [reply]
Thanks Finn,
I didn't see that part! silly me! 
Now when I am trying to see the output for that component looks like it is not recognising the different levels, but I suppose that it is because I did something wrong introducing the raster layer in the model.
Maybe it is because the information is only in the area of interest and outside of that area there is not information (not like in the gorillas example where the raster has information outside of the area of interest)
